### PR TITLE
Enhance Kasan

### PR DESCRIPTION
**Pull request by W-M-R**
## Summary
1. Add software tag kasan
2. Fixed a series of kasan detection issues
## Impact
No
## Testing
No

### CODE REVIEW

**Comment by anchao** [comment]
remove "Revert" commits in list

**Comment by W-M-R** [comment]
Error: /home/runner/work/nuttx/nuttx/nuttx/mm/kasan/hook.c:409:5: error: Mixed case identifier found
Error: /home/runner/work/nuttx/nuttx/nuttx/mm/kasan/hook.c:414:5: error: Mixed case identifier found This cannot be solved because the hwasan instrumentation function itself is case-sensitive.

**Comment by xiaoxiang781216** [comment]
> Check / check (pull_request) Error: /home/runner/work/nuttx/nuttx/nuttx/mm/kasan/hook.c:409:5: error: Mixed case identifier found Error: /home/runner/work/nuttx/nuttx/nuttx/mm/kasan/hook.c:414:5: error: Mixed case identifier found This cannot be solved because the hwasan instrumentation function itself is case-sensitive.

let's create a patch add the whitelist to tools/nxstyle.c

**Comment by W-M-R** [comment]
> > Check / check (pull_request) Error: /home/runner/work/nuttx/nuttx/nuttx/mm/kasan/hook.c:409:5: error: Mixed case identifier found Error: /home/runner/work/nuttx/nuttx/nuttx/mm/kasan/hook.c:414:5: error: Mixed case identifier found This cannot be solved because the hwasan instrumentation function itself is case-sensitive.
> 
> let's create a patch add the whitelist to tools/nxstyle.c. BTW, what's about documentation fail?

The last time the patch that added document was merged into the community, it might not have been detected.

**Review by GUIDINGLI — APPROVED** [review]

File: mm/kasan/Make.defs [hunk 1]
```diff
@@ -1,8 +1,6 @@
 ############################################################################
 # mm/kasan/Make.defs
 #
-# SPDX-License-Identifier: Apache-2.0
```
Comment (xiaoxiang781216) [inline mm/kasan/Make.defs:4]: keep

File: mm/kasan/Make.defs [hunk 2]
```diff
@@ -26,16 +24,15 @@ CSRCS += kasan.c
 
 # Disable kernel-address in mm subsystem
 
-ifeq ($(CONFIG_ARCH_TOOLCHAIN_GNU),y)
```
Comment (xiaoxiang781216) [inline mm/kasan/Make.defs:29]: why remove

File: include/nuttx/nuttx.h [hunk 3]
```diff
@@ -33,6 +33,12 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
+/* Align definitions */
+
+#ifndef ALIGN_UP
```
Comment (xiaoxiang781216) [inline include/nuttx/nuttx.h:38]: remove, use nuttx.h

File: mm/kasan/hook.c [hunk 4]
```diff
@@ -40,44 +40,50 @@
 #  define kasan_is_poisoned(addr, size) false
 #endif
 
+#ifdef CONFIG_MM_KASAN
+# define kasan_return_address(x) return_address(x)
```
Comment (xiaoxiang781216) [inline mm/kasan/hook.c:44]: why add

File: mm/kasan/generic.c [hunk 5]
```diff
@@ -206,24 +196,23 @@ bool kasan_is_poisoned(FAR const void *addr, size_t size)
...
-
-  if (size <= align)
-    {
-      return ((*p >> bit) & 1);
+      return;
     }
 
   nbit = KASAN_BITS_PER_WORD - bit % KASAN_BITS_PER_WORD;
   mask = KASAN_FIRST_WORD_MASK(bit);
-  size = ALIGN_UP(size, align);
```
Comment (xiaoxiang781216) [inline mm/kasan/generic.c]: why remove

File: Documentation/guides/index.rst [hunk 6]
```diff
@@ -19,6 +19,7 @@ Guides
   customapps.rst
   citests.rst
   zerolatencyinterrupts.rst
+  kasan.rst
```
Comment (xiaoxiang781216) [inline Documentation/guides/index.rst:22]: move to previous patch

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**